mapbox/dr-ui
# Scroll to active item in SectionedNavigation

Just like #243 this PR:

* adds a feature to automatically scroll PageLayout sidebar to the active item.
* makes slight changes (uses classnames to define conditional classes; uses React.Fragment to remove extra `div`s)
* removed `visibleSections` from the state since `filter` is the variable that gets updated by the user and the component will rebuild visibleSection as the filter changes.

Ref #243 

### Screenshots

Stage | Image
---|---
Current | ![image](https://user-images.githubusercontent.com/2180540/74359939-29cbaf80-4d92-11ea-8e3b-df960f341b32.png)
Proposed | ![image](https://user-images.githubusercontent.com/2180540/74359968-36e89e80-4d92-11ea-8f00-558ad851e7c7.png)



## How to review

1. `npm start`
2. Try these test paths by looking at the test case at the bottom called "PageLayout".
    - [x] /SectionedNavigation - The site has set "Animate a marker" as the active item. When you load the page and scroll down to the PageLayout test case, the sidebar should have scrolled to bring "Animated a marker" to the top.
    - [x] /SectionedNavigation#geocoder - The sidebar should show the heading "Geocoder" in view

I also copied this component as a test in /mapbox-gl-js-docs:

1. Open /mapbox-gl-js-docs locally, pull down the `nav-accordion-test` branch.
2. `yarn install`
3. `yarn start`
4. Test Style Spec page, compare:
    * http://localhost:8080/mapbox-gl-js/example/drag-a-marker/
    * https://docs.mapbox.com/mapbox-gl-js/example/drag-a-marker/



## QA Checklist



- [x] Accessibility score in [Chrome DevTools Audit/Lighthouse](https://developers.google.com/web/tools/lighthouse#devtools) is 100% with Lighthouse version: `5.7.0`.
  - The score should be 90:
    * `[id] attributes on the page are not unique` The PageLayout and Test cases app have this id baked in, this collision is only happening within the test cases app.
    * `Form elements do not have associated labels` Ticketed in https://github.com/mapbox/dr-ui/issues/248

Open the test cases app locally on:
